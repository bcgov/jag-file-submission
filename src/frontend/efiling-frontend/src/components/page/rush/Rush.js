/* eslint-disable react/jsx-props-no-spreading */
import React, { useState } from "react";
import PropTypes from "prop-types";
import Dropzone from "react-dropzone";
import {
  Radio,
  Callout,
  Button,
  Input,
  Textarea,
  DatePick,
  Sidecard,
} from "shared-components";
import { getSidecardData } from "../../../modules/helpers/sidecardData";
import Payment from "../../../domain/payment/Payment";

import "./Rush.scss";

const calloutText = `Please provide the date of when the direction was made,
the name of the Judge who made the direction along with any additional details
you feel are necessary.`;

const generateInputField = (input, onChange) => (
  <div className="form-child">
    <Input input={input} onChange={onChange} />
  </div>
);

export default function Rush({ payment }) {
  const input = {
    isReadOnly: false,
    styling: "bcgov-editable-white",
    isRequired: true,
  };
  const aboutCsoSidecard = getSidecardData().aboutCso;
  const csoAccountDetailsSidecard = getSidecardData().csoAccountDetails;

  const [selectedDate, setSelectedDate] = useState(new Date());
  const [showPayment, setShowPayment] = useState(false);

  if (showPayment) {
    return <Payment payment={payment} />;
  }

  return (
    <div className="ct-rush page">
      <div className="content col-md-8">
        <h1>Request Rush/Urgent Submission</h1>
        <p>
          You must provide a reason and contact information for your request for
          rush/urgent submission to be considered. My reason for requesting
          urgent (rush) filing is:
        </p>
        <Radio
          id="rule-85"
          name="rush-reason"
          label="The attached application is made under Rule 8-5 (1) SCR."
        />
        <Radio
          id="court-directed"
          name="rush-reason"
          label="The court directed that the order be processed on an urgent basis."
        />
        <Radio id="other" name="rush-reason" label="Other (please explain)." />
        <Callout text={calloutText} />
        <DatePick
          label="Date"
          isRequired
          selectedDate={selectedDate}
          setSelectedDate={setSelectedDate}
        />
        <br />
        <Textarea
          id="detailed-reason"
          isRequired
          label="Clear and detailed reason(s)"
        />
        <p>
          I understand the registry has the discretion to reject this request.
        </p>
        <div className="form-parent">
          {generateInputField({
            ...input,
            label: "Last Name",
            id: "lastName",
          })}
          {generateInputField({
            ...input,
            label: "Organization",
            id: "org",
            isRequired: false,
          })}
        </div>
        <div className="form-parent">
          {generateInputField({
            ...input,
            label: "First Name",
            id: "firstName",
          })}
          {generateInputField({
            ...input,
            label: "Contact Country",
            id: "country",
            isRequired: false,
          })}
        </div>
        <div className="form-parent">
          {generateInputField({
            ...input,
            label: "Phone Number",
            id: "phoneNumber",
          })}
        </div>
        <br />
        <Dropzone>
          {({ getRootProps, getInputProps }) => (
            <div
              data-testid="dropdownzone"
              {...getRootProps({ className: "dropzone-outer" })}
            >
              <div className="dropzone-inner-box">
                <input {...getInputProps()} />
                <span>
                  <h2 className="text-center-alignment">
                    Drag and drop or&nbsp;
                    <span className="file-href">choose documents</span>
                  </h2>
                  <div>
                    <span>
                      to upload supporting documents from counsel that explain
                      the reasons for an urgent/rush order.
                    </span>
                    <br />
                    <span>
                      Please note that documents uploaded on this screen will
                      not be processed into your eFile.
                    </span>
                  </div>
                </span>
              </div>
            </div>
          )}
        </Dropzone>
        <br />
        <section className="buttons pt-2">
          <Button
            label="Cancel Request"
            onClick={() => setShowPayment(true)}
            styling="bcgov-normal-white btn"
          />
          <Button label="Submit Request" styling="bcgov-normal-blue btn" />
        </section>
      </div>
      <div className="sidecard">
        <Sidecard sideCard={csoAccountDetailsSidecard} />
        <Sidecard sideCard={aboutCsoSidecard} />
      </div>
    </div>
  );
}

Rush.propTypes = {
  payment: PropTypes.object.isRequired,
};
