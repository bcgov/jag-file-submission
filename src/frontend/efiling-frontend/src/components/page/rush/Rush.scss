.ct-rush {
  .form-parent {
    display: flex;
    justify-content: space-between;
    padding-bottom: 15px;
  }

  .form-child {
    width: 45%;
  }

  #detailed-reason {
    width: 100%;
    height: 200px;
    margin-bottom: 40px;
  }

  .dropzone-outer {
    border: 1px dashed #707070;
    height: 200px;
    display: flex;
    justify-content: center;
  }

  .bcgov-dropdown-wrapper {
    width: 100%;
  }

  .bcgov-dropdown {
    border-radius: 4px;
    border-color: darkgrey;
  }

  .bcgov-dropdown-label {
    font-weight: bold;
    margin-bottom: 8px !important;
  }

  .bcgov-editable-white {
    height: 45px;
    border-style: solid;
    border-color: darkgrey;
    border-radius: 4px;
  }

  .red {
    color: red;
    font-weight: 200;
  }

  @media (max-width: 576px) {
    .dropzone-outer {
      border: 1px dashed #707070;
      height: 250px;
      display: flex;
      justify-content: center;
    }
  }
}
