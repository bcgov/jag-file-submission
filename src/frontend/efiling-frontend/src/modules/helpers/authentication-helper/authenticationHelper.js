const jwt = require("jsonwebtoken");

/**
 * Encodes/signs a payload with JWT into a JSON Web Token string.
 * @param payload Payload to sign, could be an literal, buffer or string
 */
export function encodeJWT(payload) {
  return jwt.sign(payload, "secret");
}

/**
 * Decodes a JWT token, returning the payload, or null if no payload can be found.
 * @param token a JWT token string to decode
 */
export function decodeJWT(token) {
  const decoded = jwt.decode(token, { complete: true });

  if (decoded && decoded.payload) return decoded.payload;
  return null;
}

export function getJWTData() {
  const token = localStorage.getItem("jwt");

  // get the decoded payload and header
  return decodeJWT(token);
}

export function generateJWTToken(payload) {
  return jwt.sign(payload, "secret");
}

/**
 * Returns the identity provider of the current user.
 */
export function getIdentityProviderAlias() {
  const token = getJWTData();
  return token.identityProviderAlias;
}
