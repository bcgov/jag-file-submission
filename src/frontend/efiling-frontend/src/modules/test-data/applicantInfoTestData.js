const applicantInfo = {
  bceid: "bobross42",
  firstName: "Bob",
  lastName: "Ross",
  email: "bob.ross@example.com",
};

export function getApplicantInfo() {
  return applicantInfo;
}
