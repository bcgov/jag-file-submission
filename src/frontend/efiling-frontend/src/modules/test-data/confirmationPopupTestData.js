/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";

const onButtonClick = () => {};

const mainButton = {
  label: "Cancel",
  styling: "bcgov-normal-white btn",
  onClick: onButtonClick,
};

const confirmButton = {
  label: "Yes, cancel",
  styling: "bcgov-normal-blue btn bcgov-consistent-width",
  onClick: onButtonClick,
};

const cancelButton = {
  label: "No, dont cancel",
  styling: "bcgov-normal-white btn bcgov-consistent-width",
  onClick: onButtonClick,
};

const body = () => (
  <>
    <p>Your files will not be submitted.</p>
    <p>
      You will be returned to:
      <br />
      <b>Original</b> website
    </p>
  </>
);

export function getTestData() {
  const modal = {
    show: false,
    title: "title",
    body,
  };

  return {
    modal,
    mainButton,
    confirmButton,
    cancelButton,
  };
}
