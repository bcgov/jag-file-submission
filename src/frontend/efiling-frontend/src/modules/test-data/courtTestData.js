const court = {
  locationDescription: "Court location",
  fileNumber: "Court file number",
  levelDescription: "Level",
  classDescription: "Class",
  level: "P",
  courtClass: "F",
  location: "Kelowna Law Courts",
};

export function getCourtData() {
  return court;
}
