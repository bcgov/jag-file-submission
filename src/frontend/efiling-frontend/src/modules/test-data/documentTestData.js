const documents = [
  {
    documentProperties: {
      name: "file name 1",
      type: "file type",
    },
    description: "file description 1",
    statutoryFeeAmount: 40,
    isAmendment: null,
    isSupremeCourtScheduling: null,
  },
  {
    documentProperties: {
      name: "file name 2",
      type: "file type",
    },
    description: "file description 2",
    statutoryFeeAmount: 0,
    isAmendment: false,
    isSupremeCourtScheduling: true,
  },
];

export function getDocumentsData() {
  return documents;
}
