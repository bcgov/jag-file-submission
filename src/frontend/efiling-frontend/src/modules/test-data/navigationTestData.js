const navigationUrls = {
  cancel: "cancelurl.com",
  success: "successurl.com",
  error: "",
};

export function getNavigationData() {
  return navigationUrls;
}
