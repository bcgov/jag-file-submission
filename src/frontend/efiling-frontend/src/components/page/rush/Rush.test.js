// TODO: remove ignoring Rush.js from coverage report
import React from "react";
import { render, fireEvent, getByText } from "@testing-library/react";
import { getDocumentsData } from "../../../modules/test-data/documentTestData";
import { getTestData } from "../../../modules/test-data/confirmationPopupTestData";
import { getCourtData } from "../../../modules/test-data/courtTestData";
import { generateJWTToken } from "../../../modules/helpers/authentication-helper/authenticationHelper";

import Rush from "./Rush";

describe("Rush Component", () => {
  let realDate;
  const confirmationPopup = getTestData();
  const submissionId = "abc123";
  const courtData = getCourtData();
  const files = getDocumentsData();
  const submissionFee = 25.5;

  const payment = {
    confirmationPopup,
    submissionId,
    courtData,
    files,
    submissionFee,
  };

  const token = generateJWTToken({
    preferred_username: "username@bceid",
    realm_access: {
      roles: ["rush_flag"],
    },
  });
  localStorage.setItem("jwt", token);
  sessionStorage.setItem("csoBaseUrl", "https://dev.justice.gov.bc.ca/cso");
  window.scrollTo = jest.fn();

  test("Matches the snapshot", () => {
    const currentDate = new Date("2019-05-14T11:01:58.135Z");
    realDate = Date;
    global.Date = class extends Date {
      constructor() {
        return currentDate;
      }
    };

    const { asFragment } = render(<Rush payment={payment} />);

    expect(asFragment()).toMatchSnapshot();

    global.Date = realDate;
  });

  test("Clicking on cancel request takes user back to payment page", () => {
    const { container, asFragment } = render(<Rush payment={payment} />);

    const button = getByText(container, "Cancel Request");

    fireEvent.click(button);

    expect(asFragment()).toMatchSnapshot();
  });
});
