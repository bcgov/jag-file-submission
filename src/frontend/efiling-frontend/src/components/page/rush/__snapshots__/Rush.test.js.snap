// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rush Component Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ct-rush page"
  >
    <div
      class="content col-md-8"
    >
      <h1>
        Rush Details
      </h1>
      <h2>
        Submitting on a rush basis
      </h2>
      <br />
      <p>
        If you wish to request that this package be submitted on an urgent (rush) basis, 
        <br />
        you must provide a reason and contact information for your request to be considered.
      </p>
      <br />
      <h3>
        My reason for requesting urgent (rush) filing is:
      </h3>
      <br />
      <label
        class="bcgov-radio"
        for="rule-85"
      >
        The attached application is made under Rule 8-5 (1) SCR.
        <input
          id="rule-85"
          name="rush-reason"
          type="radio"
        />
        <span
          class="bcgov-dot"
        />
      </label>
      <label
        class="bcgov-radio"
        for="court-directed"
      >
        The court directed that the order be processed on an urgent basis.
        <input
          id="court-directed"
          name="rush-reason"
          type="radio"
        />
        <span
          class="bcgov-dot"
        />
      </label>
      <label
        class="bcgov-radio"
        for="other"
      >
        Other (please explain).
        <input
          id="other"
          name="rush-reason"
          type="radio"
        />
        <span
          class="bcgov-dot"
        />
      </label>
      <br />
      <section
        class="buttons pt-2"
      >
        <button
          class="bcgov-button bcgov-normal-white btn"
          data-testid=""
          type="button"
        >
          Cancel
        </button>
        <button
          class="bcgov-button bcgov-normal-blue btn"
          data-testid=""
          disabled=""
          type="button"
        >
          Continue
        </button>
      </section>
    </div>
    <div
      class="sidecard"
    >
      <div
        class="bcgov-wide-dashboard-spacing"
        style="position: relative;"
      >
        <div
          class="col-12 pt-2"
          style="position: relative;"
        >
          <p />
          <section
            class="bcgov-bluegrey-container"
            id="rushSubmissionCard"
          >
            <div
              class="bcgov-container-background bcgov-bluegrey-heading"
            >
              <h2
                class="bcgov-heading-style"
              >
                <div
                  class="bcgov-side-card-row"
                >
                  <div
                    class="bcgov-round-icon-wrapper"
                  >
                    <svg
                      class="bcgov-side-card-icon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M15 1H9v2h6V1zm-4 13h2V8h-2v6zm8.03-6.61l1.42-1.42c-.43-.51-.9-.99-1.41-1.41l-1.42 1.42A8.962 8.962 0 0012 4c-4.97 0-9 4.03-9 9s4.02 9 9 9a8.994 8.994 0 007.03-14.61zM12 20c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
                      />
                    </svg>
                  </div>
                  <div
                    class="bcgov-side-card-title"
                  >
                    About Rush Documents
                  </div>
                </div>
              </h2>
            </div>
            <div
              class="bcgov-bluegrey-content"
            >
              <span
                class="bcgov-content-style"
              >
                <p>
                  When directed by the court, an order will be processed on an urgent(rush) basis. The registry will consider specific reasons for processing an order on an urgent(rush) basis.
                </p>
                <p>
                  <b>
                    Only request processing on an urgent(rush) basis in exceptional circumstances.
                  </b>
                </p>
                <span
                  class="file-href"
                >
                  Learn more about rush processing.
                </span>
              </span>
            </div>
          </section>
        </div>
      </div>
      <div
        class="bcgov-wide-dashboard-spacing"
        style="position: relative;"
      >
        <div
          class="col-12 pt-2"
          style="position: relative;"
        >
          <p />
          <section
            class="bcgov-bluegrey-container"
            id="csoAccountDetailsCard"
          >
            <div
              class="bcgov-container-background bcgov-bluegrey-heading"
            >
              <h2
                class="bcgov-heading-style"
              >
                <div
                  class="bcgov-side-card-row"
                >
                  <div
                    class="bcgov-round-icon-wrapper"
                  >
                    <svg
                      class="bcgov-side-card-icon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
                      />
                    </svg>
                  </div>
                  <div
                    class="bcgov-side-card-title"
                  >
                    Your CSO Account
                  </div>
                </div>
              </h2>
            </div>
            <div
              class="bcgov-bluegrey-content"
            >
              <span
                class="bcgov-content-style"
              >
                <p>
                  CSO account 
                  <strong />
                   is linked to your Basic BCeID account 
                  <strong>
                    username
                  </strong>
                   and will be used to file documents. 
                  <a
                    href="undefined/accounts/editProfile.do"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    View your CSO account details
                  </a>
                  .
                </p>
              </span>
            </div>
          </section>
        </div>
      </div>
      <div
        class="bcgov-wide-dashboard-spacing"
        style="position: relative;"
      >
        <div
          class="col-12 pt-2"
          style="position: relative;"
        >
          <p />
          <section
            class="bcgov-bluegrey-container"
            id="aboutCsoCard"
          >
            <div
              class="bcgov-container-background bcgov-bluegrey-heading"
            >
              <h2
                class="bcgov-heading-style"
              >
                <div
                  class="bcgov-side-card-row"
                >
                  <div
                    class="bcgov-round-icon-wrapper"
                  >
                    <svg
                      class="bcgov-side-card-icon"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0V0z"
                        fill="none"
                      />
                      <path
                        d="M11 7h2v2h-2V7zm0 4h2v6h-2v-6zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                      />
                    </svg>
                  </div>
                  <div
                    class="bcgov-side-card-title"
                  >
                    About E-File Submission
                  </div>
                </div>
              </h2>
            </div>
            <div
              class="bcgov-bluegrey-content"
            >
              <span
                class="bcgov-content-style"
              >
                <p>
                  E-File Submission is a service to help you securely and electronically file documents with the Courts of British Columbia via Court Services Online. 
                  <a
                    href="undefined/about/index.do"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Learn more about CSO
                  </a>
                  .
                </p>
              </span>
            </div>
          </section>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
